sk0-back/src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { Strategy, ExtractJwt } from 'passport-jwt';
+import { JwtPayload } from './jwt-payload.interface';
+import { UsersService } from 'src/users/services/users.service';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy) {
+  userService: UsersService;
```

Comment:
This `UserService`, can you please elaborate where this thing is coming from? Is it injected as a dependency?

Reply:
Can you please add a few happy and unhappy case scenarios as tests? This thing is unused or should be failing. :|

Reply:
it would be useful to inject the UserService into the constructor to access users with their email within the validate method

Reply:
yes it's for validating the users but I'll need some help figuring out the dependencies because it's a mess right now